TheEthicalHackerCup/Assets/AttackMinigames/Ransomware/Scripts/RansomManager.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using UnityEngine;
+
+public class RansomManager : MonoBehaviour
```

Comment:
I'm curious why you moved away from a singleton approach. I'm not sure if you can guarantee that the same object instance of this will always be used otherwise and RansomUi.cs getting this via `FindObjectOfType` gets the first available one if multiple instances exist

Reply:
Singleton was alright, but I wasn't sure if using it would persist between plays. that could cause annoyances.

I think there will only ever be one so we good

Reply:
Any singleton data persistence between plays can be avoided by implementing an `Initialize()` method that sets all fields to defaults, and then calling that at the start of your scene

Your updated approch with `FindObjectOfType<>` will work, but Unity docs says that it is slower than a singleton approach so if there are any noticeable performance issues, you may need to change
https://docs.unity3d.com/ScriptReference/Object.FindObjectOfType.html